props/saber_BC_buttons.h

```diff
@@ -383,34 +1158,253 @@ EFFECT(voldown);    // for decrease volume
+  const int max_volume_ = VOLUME;
+  const int min_volume_ = VOLUME * 0.10;
+  float initial_volume_ = 0.0;
+  int initial_percentage_ = 0;
+  int percentage_ = 0;
+  bool announced_ = false;
+
+  int steps_per_revolution() override {
+    return 12;  // adjust for sensitivity
+  }
 
-class DelayTimer {
-public:
-    DelayTimer() : triggered_(false), trigger_time_(0), duration_(0) {}
+  void mode_activate(bool onreturn) override {
+    PVLOG_NORMAL << "** Enter Volume Menu\n";
+    initial_volume_ = dynamic_mixer.get_volume();
+    initial_percentage_ = round((initial_volume_ / max_volume_) * 10) * 10;
+   SaberBase::DoEffect(EFFECT_VOLUME_LEVEL, 0);
+    mode::getSL<SPEC>()->SayEnterVolumeMenu();
+    SPEC::SteppedMode::mode_activate(onreturn);
+  }
+
+  void mode_deactivate() override {
+    if (percentage_ <= 10) {
+      mode::getSL<SPEC>()->SayMinimumVolume();
+    } else if (percentage_ >=100) {
+      mode::getSL<SPEC>()->SayMaximumVolume();
+    } else {
+      mode::getSL<SPEC>()->SayWhole(percentage_);
+      mode::getSL<SPEC>()->SayPercent();
+    }
+    mode::getSL<SPEC>()->SayVolumeMenuEnd();
+    SPEC::SteppedMode::mode_deactivate();
+  }
 
-    void trigger(uint32_t duration) {
-        triggered_ = true;
-        trigger_time_ = millis();
-        duration_ = duration;
+  void next() override {
+    int current_volume_ = dynamic_mixer.get_volume();
+    if (current_volume_ < max_volume_) {
+      current_volume_ += max_volume_ * 0.10;
+      dynamic_mixer.set_volume(current_volume_);
+      mode::getSL<SPEC>()->SayVolumeUp();
+      announced_ = false;
+    } else if (!announced_) {
+      QuickMaxVolume();
+      announced_ = true;
     }
+  }
 
-    bool timerCheck() {
-        if (!triggered_) return false;
-        if (millis() - trigger_time_ > duration_) {
-            triggered_ = false;  // Reset the timer flag
-            return true;  // Timer has elapsed
-        }
-        return false;  // Timer is still running
+  void prev() override {
+    int current_volume_ = dynamic_mixer.get_volume();
+    if (current_volume_ > min_volume_) {
+      current_volume_ -= max_volume_ * 0.10;
+      dynamic_mixer.set_volume(current_volume_);
+      mode::getSL<SPEC>()->SayVolumeDown();
+      announced_ = false;
+    } else if (!announced_) {
+      QuickMinVolume();
+      announced_ = true;
+    }
+  }
+
+  void QuickMaxVolume() {
+    dynamic_mixer.set_volume(max_volume_);
+    PVLOG_NORMAL << "** Maximum Volume\n";
+    mode::getSL<SPEC>()->SayMaximumVolume();
+  }
+
+  void QuickMinVolume() {
+    dynamic_mixer.set_volume(min_volume_);
+    PVLOG_NORMAL << "** Minimum Volume\n";
+    mode::getSL<SPEC>()->SayMinimumVolume();
+  }
+
+  void update() override {
+    float volume = dynamic_mixer.get_volume();
+    percentage_ = round((volume / max_volume_) * 10) * 10;
+    SaberBase::DoEffect(EFFECT_VOLUME_LEVEL, 0);
+    // Overridden to substitute the tick sound
+    this->say_time_ = Cyclint<uint32_t>(millis()) + 1;
```

Comment:
I think you can skip setting say_time_ since there is no say() function in this class.

Reply:
Indeed. Done.